packages/runtime/src/next/client.ts

```diff
@@ -202,6 +208,32 @@ export class Makeswift {
     return json
   }
 
+  async getPage(
+    pathname: string,
+    { locale: localeInput }: { locale?: string } = {},
+  ): Promise<GetPageResult | null> {
+    const siteVersion = MakeswiftSiteVersion.Live
+
+    const searchParams = new URLSearchParams()
+    if (localeInput) {
+      searchParams.set('locale', localeInput)
+    }
+
+    const response = await this.fetch(
+      `v1/pages/${encodeURIComponent(pathname)}/meta?${searchParams.toString()}`,
+      { headers: { 'Makeswift-Site-Version': siteVersion } },
+    )
+
+    if (!response.ok) {
+      console.error('Failed to get page', await response.json())
+      throw new Error(`Failed to get page with error: "${response.statusText}"`)
+    }
+
+    const json: GetPageResult = await response.json()
```

Comment:
What do you think about using Zod to validate and transform the response?

Reply:
Updated